modules/aws/cac/cac-startup.sh.tmpl

```diff
@@ -15,11 +15,22 @@ log() {
 }
 
 get_credentials() {
-    log "Not using encryption"
+    if [[ -z "${customer_master_key_id}" ]]; then
+        log "Not using encryption"
 
-    PCOIP_REGISTRATION_CODE=${pcoip_registration_code}
-    SERVICE_ACCOUNT_PASSWORD=${service_account_password}
-    CAC_TOKEN=${cac_token}
+        PCOIP_REGISTRATION_CODE=${pcoip_registration_code}
+        SERVICE_ACCOUNT_PASSWORD=${service_account_password}
+        CAC_TOKEN=${cac_token}
+
+    else
+        log "Using encryption key ${customer_master_key_id}"
+
+        apt install -y awscli
+
+        PCOIP_REGISTRATION_CODE=$(aws kms decrypt --region ${aws_region} --ciphertext-blob fileb://<(echo "${pcoip_registration_code}" | base64 -d) --output text --query Plaintext | base64 -d)
+        SERVICE_ACCOUNT_PASSWORD=$(aws kms decrypt --region ${aws_region} --ciphertext-blob fileb://<(echo "${service_account_password}" | base64 -d) --output text --query Plaintext | base64 -d)
+        CAC_TOKEN=$(aws kms decrypt --region ${aws_region} --ciphertext-blob fileb://<(echo "${cac_token}" | base64 -d) --output text --query Plaintext | base64 -d)
```

Comment:
This decryption is via curl POST using access token for GCP, should we convert that one to use gcloud?

Reply:
we could, but if it works now, we don't have to change it unless there are real benefits.